odl/test/discr/lp_discr_test.py

```diff
@@ -1267,17 +1267,14 @@ def test_norm_rectangle(exponent):
         assert discr_testfunc.norm() <= 1  # Max at boundary not hit
     else:
         true_norm = ((1 + 2 * p) * (1 + 3 * p) / 2) ** (-1 / p)
-        assert almost_equal(discr_testfunc.norm(), true_norm, places=2)
+        assert discr_testfunc.norm() == pytest.approx(true_norm, rel=1e-2)
 
 
 def test_norm_rectangle_boundary(odl_tspace_impl, exponent):
     # Check the constant function 1 in different situations regarding the
     # placement of the outermost grid points.
     impl = odl_tspace_impl
 
-    if exponent == float('inf'):
-        pytest.xfail('inf-norm not implemented in CUDA')
```

Comment:
Just curious, why did we have `xfail` on this one before and what has changed now?

Reply:
The inf norm wasn't implemented on CUDA, and I guess we weren't (i.e. I wasn't) smart enough to xfail only in that case. Now we have no CUDA backend, and the new one will support it, so this restriction doesn't make any more sense.